lib/Frontend/ModuleInterfaceSupport.cpp

```diff
@@ -621,7 +623,7 @@ class InheritedProtocolCollector {
       return;
 
     // Skip SPI extensions in the public interface.
-    if (!printOptions.PrintSPIs && extension->isSPI())
+    if (printOptions.InterfaceContentMode == InterfaceMode::Public && extension->isSPI())
```

Comment:
We could probably put `.InterfaceContentMode == InterfaceMode::Public` into a helper method on `printOptions` to make this simpler, since it appears a lot.